{"metadata": {"kernelspec": {"name": "python3", "display_name": "Python 3", "language": "python"}, "language_info": {"name": "python", "version": "3.7.3", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "pygments_lexer": "ipython3", "nbconvert_exporter": "python", "file_extension": ".py"}}, "nbformat_minor": 4, "nbformat": 4, "cells": [{"cell_type": "markdown", "source": "## Getting GIAB data and selectings regions covering MHC region", "metadata": {}}, {"cell_type": "markdown", "source": "The data was downloaded from the following links:\n\nPacBio CCS 15Kb Sequel I - ftp://ftp-trace.ncbi.nlm.nih.gov/giab/ftp/data/AshkenazimTrio/HG002_NA24385_son/PacBio_CCS_15kb/alignment/HG002.Sequel.15kb.pbmm2.hs37d5.whatshap.haplotag.RTG.10x.trio.bam \n\nPacBio CCS 11Kb Sequel II - ftp://ftp-trace.ncbi.nlm.nih.gov/giab/ftp/data/AshkenazimTrio/HG002_NA24385_son/PacBio_SequelII_CCS_11kb/HG002.SequelII.pbmm2.hs37d5.whatshap.haplotag.RTG.10x.trio.bam \n\nONT \"ultralong\" - ftp://ftp-trace.ncbi.nlm.nih.gov/giab/ftp/data/AshkenazimTrio/HG002_NA24385_son/Ultralong_OxfordNanopore/final/ultra-long-ont_hs37d5_phased_reheader.bam\n", "metadata": {}}, {"cell_type": "code", "source": "%%bash\n# wget --quiet ftp://ftp-trace.ncbi.nlm.nih.gov/giab/ftp/data/AshkenazimTrio/HG002_NA24385_son/PacBio_CCS_15kb/alignment/HG002.Sequel.15kb.pbmm2.hs37d5.whatshap.haplotag.RTG.10x.trio.bam &\n# wget --quiet ftp://ftp-trace.ncbi.nlm.nih.gov/giab/ftp/data/AshkenazimTrio/HG002_NA24385_son/PacBio_SequelII_CCS_11kb/HG002.SequelII.pbmm2.hs37d5.whatshap.haplotag.RTG.10x.trio.bam &\n# wget --quiet ftp://ftp-trace.ncbi.nlm.nih.gov/giab/ftp/data/AshkenazimTrio/HG002_NA24385_son/Ultralong_OxfordNanopore/final/ultra-long-ont_hs37d5_phased_reheader.bam &\n# wget --quiet ftp://ftp-trace.ncbi.nlm.nih.gov/giab/ftp/data/AshkenazimTrio/HG002_NA24385_son/PacBio_CCS_15kb/alignment/HG002.Sequel.15kb.pbmm2.hs37d5.whatshap.haplotag.RTG.10x.trio.bam.bai &\n# wget --quiet ftp://ftp-trace.ncbi.nlm.nih.gov/giab/ftp/data/AshkenazimTrio/HG002_NA24385_son/PacBio_SequelII_CCS_11kb/HG002.SequelII.pbmm2.hs37d5.whatshap.haplotag.RTG.10x.trio.bam.bai &\n# wget --quiet ftp://ftp-trace.ncbi.nlm.nih.gov/giab/ftp/data/AshkenazimTrio/HG002_NA24385_son/Ultralong_OxfordNanopore/final/ultra-long-ont_hs37d5_phased_reheader.bam.bai &", "metadata": {"trusted": true}, "execution_count": 1, "outputs": []}, {"cell_type": "code", "source": "%%bash\ndx download --recursive /GIAB_BAM_files/", "metadata": {"trusted": true}, "execution_count": null, "outputs": []}, {"cell_type": "code", "source": "%%bash\nln -s GIAB_BAM_files/* .", "metadata": {"trusted": true}, "execution_count": null, "outputs": []}, {"cell_type": "markdown", "source": "NOTE: had to use the following conda environment and whatshap build to have whatshap haplotag --output-haplotag-list work:\n\nconda create -y -n whatshap minimap2 samtools bcftools pysam -c bioconda\n\nconda activate whatshap; pip install git+https://bitbucket.org/whatshap/whatshap@split\n\ncommit id: 4f994e8\n\nPulling software", "metadata": {}}, {"cell_type": "code", "source": "%%bash\nconda install -y minimap2 samtools bcftools pysam bamtools -c bioconda\napt-get install -y build-essential pigz\npip install git+https://bitbucket.org/whatshap/whatshap@split", "metadata": {"trusted": true}, "execution_count": null, "outputs": []}, {"cell_type": "markdown", "source": "For PacBio CCS Sequel 15Kb, Sequel II 11Kb, and ONT \"ultra-long\" the reads covering the MHC were selected using the following commands (with bam file name updated for each data) for haplotype 1 and haplotype 2:\n\nThis script was run with a directory structure that includes subdirectories of \"reads\", cram\", \"ref\", \"vcf\", \"whatshap\"", "metadata": {}}, {"cell_type": "code", "source": "%%writefile filter_script.json\n{ \"filters\" : [ { \"id\" : \"inHP1\", \"tag\" : \"HP:1\" }, { \"id\" : \"inHP2\", \"tag\" : \"HP:2\" } ], \"rule\" : \"!(inHP1 | inHP2)\" }", "metadata": {"trusted": true}, "execution_count": 1, "outputs": [{"name": "stdout", "text": "Writing filter_script.json\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "%%bash\nsamtools view -bh HG002.Sequel.15kb.pbmm2.hs37d5.whatshap.haplotag.RTG.10x.trio.bam 6:28477797-33448354 | \\\n    bamtools filter -in stdin -tag HP:1 | \\\n        samtools bam2fq - | \\\n        bgzip -c > HG002.Sequel.15kb.pbmm2.hs37d5.whatshap.haplotag.RTG.10x.trio.MHConly.HP1.fastq.gz \n\nsamtools view -bh HG002.Sequel.15kb.pbmm2.hs37d5.whatshap.haplotag.RTG.10x.trio.bam 6:28477797-33448354 | \\\n    bamtools filter -in stdin -tag HP:2 | \\\n        samtools bam2fq - | \\\n        bgzip -c > HG002.Sequel.15kb.pbmm2.hs37d5.whatshap.haplotag.RTG.10x.trio.MHConly.HP2.fastq.gz\n\nsamtools view -bh HG002.Sequel.15kb.pbmm2.hs37d5.whatshap.haplotag.RTG.10x.trio.bam 6:28477797-33448354 | \\\n    bamtools filter -in stdin -script filter_script.json | \\\n    samtools bam2fq - | \\\n    bgzip -c > HG002.Sequel.15kb.pbmm2.hs37d5.whatshap.haplotag.RTG.10x.trio.MHConly.UnPartitioned.fastq.gz", "metadata": {"trusted": true}, "execution_count": 2, "outputs": [{"name": "stderr", "text": "[W::hts_idx_load2] The index file is older than the data file: HG002.Sequel.15kb.pbmm2.hs37d5.whatshap.haplotag.RTG.10x.trio.bam.bai\n[M::bam2fq_mainloop] discarded 0 singletons\n[M::bam2fq_mainloop] processed 5210 reads\n[W::hts_idx_load2] The index file is older than the data file: HG002.Sequel.15kb.pbmm2.hs37d5.whatshap.haplotag.RTG.10x.trio.bam.bai\n[M::bam2fq_mainloop] discarded 0 singletons\n[M::bam2fq_mainloop] processed 4932 reads\n[W::hts_idx_load2] The index file is older than the data file: HG002.Sequel.15kb.pbmm2.hs37d5.whatshap.haplotag.RTG.10x.trio.bam.bai\n[M::bam2fq_mainloop] discarded 0 singletons\n[M::bam2fq_mainloop] processed 1126 reads\n", "output_type": "stream"}]}, {"cell_type": "markdown", "source": "filter_script.json:\n{\n    \"filters\" : [\n    { \"id\" : \"inHP1\", \"tag\" : \"HP:1\" },\n    { \"id\" : \"inHP2\", \"tag\" : \"HP:2\" }\n    ],\n    \"rule\" : \"!(inHP1 | inHP2)\"\n}", "metadata": {}}, {"cell_type": "code", "source": "%%bash\nsamtools view -bh HG002.SequelII.pbmm2.hs37d5.whatshap.haplotag.RTG.10x.trio.bam 6:28477797-33448354 | \\\n    bamtools filter -in stdin -tag HP:1 | \\\n        samtools bam2fq - | \\\n        bgzip -c > HG002.SequelII.pbmm2.hs37d5.whatshap.haplotag.RTG.10x.trio.MHConly.HP1.fastq.gz \n\nsamtools view -bh HG002.SequelII.pbmm2.hs37d5.whatshap.haplotag.RTG.10x.trio.bam 6:28477797-33448354 | \\\n    bamtools filter -in stdin -tag HP:2 | \\\n        samtools bam2fq - | \\\n        bgzip -c > HG002.SequelII.pbmm2.hs37d5.whatshap.haplotag.RTG.10x.trio.MHConly.HP2.fastq.gz\n\nsamtools view -bh HG002.SequelII.pbmm2.hs37d5.whatshap.haplotag.RTG.10x.trio.bam 6:28477797-33448354 | \\\n    bamtools filter -in stdin -script filter_script.json | \\\n    samtools bam2fq - | \\\n    bgzip -c > HG002.SequelII.pbmm2.hs37d5.whatshap.haplotag.RTG.10x.trio.MHConly.UnPartitioned.fastq.gz", "metadata": {"trusted": true}, "execution_count": 3, "outputs": [{"name": "stderr", "text": "[W::hts_idx_load2] The index file is older than the data file: HG002.SequelII.pbmm2.hs37d5.whatshap.haplotag.RTG.10x.trio.bam.bai\n[M::bam2fq_mainloop] discarded 0 singletons\n[M::bam2fq_mainloop] processed 6764 reads\n[W::hts_idx_load2] The index file is older than the data file: HG002.SequelII.pbmm2.hs37d5.whatshap.haplotag.RTG.10x.trio.bam.bai\n[M::bam2fq_mainloop] discarded 0 singletons\n[M::bam2fq_mainloop] processed 6355 reads\n[W::hts_idx_load2] The index file is older than the data file: HG002.SequelII.pbmm2.hs37d5.whatshap.haplotag.RTG.10x.trio.bam.bai\n[M::bam2fq_mainloop] discarded 0 singletons\n[M::bam2fq_mainloop] processed 1535 reads\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "%%bash\nsamtools view -bh ultra-long-ont_hs37d5_phased_reheader.bam 6:28477797-33448354 | \\\n    bamtools filter -in stdin -tag HP:1 | \\\n        samtools bam2fq - | \\\n        bgzip -c > HG002.ultra-long-ont_hs37d5_phased_reheader.HP1.fastq.gz\n\nsamtools view -bh ultra-long-ont_hs37d5_phased_reheader.bam 6:28477797-33448354 | \\\n    bamtools filter -in stdin -tag HP:2 | \\\n        samtools bam2fq - | \\\n        bgzip -c > HG002.ultra-long-ont_hs37d5_phased_reheader.HP2.fastq.gz\n\nsamtools view -bh ultra-long-ont_hs37d5_phased_reheader.bam 6:28477797-33448354 | \\\n    bamtools filter -in stdin -script filter_script.json | \\\n    samtools bam2fq - | \\\n    bgzip -c > HG002.ultra-long-ont_hs37d5_phased_reheader.MHConly.UnPartitioned.fastq.gz", "metadata": {"trusted": true}, "execution_count": 4, "outputs": [{"name": "stderr", "text": "[M::bam2fq_mainloop] discarded 0 singletons\n[M::bam2fq_mainloop] processed 5678 reads\n[M::bam2fq_mainloop] discarded 0 singletons\n[M::bam2fq_mainloop] processed 5650 reads\n[M::bam2fq_mainloop] discarded 0 singletons\n[M::bam2fq_mainloop] processed 4180 reads\n", "output_type": "stream"}]}, {"cell_type": "markdown", "source": "## Tobias's haplotype binning of reads", "metadata": {}}, {"cell_type": "markdown", "source": "Getting the input files", "metadata": {}}, {"cell_type": "code", "source": "%%bash\nwget --quiet ftp://ftp-trace.ncbi.nlm.nih.gov/giab/ftp/data/AshkenazimTrio/analysis/10XGenomics_ChromiumGenome_LongRanger2.2_Supernova2.0.1_04122018/GRCh37/NA24385_300G/NA24385.GRCh37.phased_variants.vcf.gz\nwget --quiet ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/reference/phase2_reference_assembly_sequence/hs37d5.fa.gz", "metadata": {"trusted": true}, "execution_count": 5, "outputs": []}, {"cell_type": "code", "source": "%%bash\nmkdir -p ref\nmv hs37d5.fa.gz ref", "metadata": {"trusted": true}, "execution_count": 6, "outputs": []}, {"cell_type": "markdown", "source": "Merging FASTQs again", "metadata": {}}, {"cell_type": "code", "source": "%%bash\napt-get install pigz\nmkdir -p reads\nzcat HG002.ultra-long-ont_hs37d5_phased_reheader.*.fastq.gz | pigz > reads/HG002_ultra-long-ont_hs37d5_phased_reheader.MHConly.fastq.gz\nzcat HG002.Sequel.15kb.pbmm2.hs37d5.whatshap.haplotag.RTG.10x.trio.MHConly.*.fastq.gz | pigz > reads/HG002.Sequel.15kb.pbmm2.hs37d5.whatshap.haplotag.RTG.10x.trio.MHConly.fastq.gz\nzcat HG002.SequelII.pbmm2.hs37d5.whatshap.haplotag.RTG.10x.trio.MHConly.*.fastq.gz | pigz >> reads/HG002.Sequel.15kb.pbmm2.hs37d5.whatshap.haplotag.RTG.10x.trio.MHConly.fastq.gz", "metadata": {"scrolled": true, "trusted": true}, "execution_count": 7, "outputs": [{"name": "stdout", "text": "Reading package lists...\nBuilding dependency tree...\nReading state information...\npigz is already the newest version (2.4-1).\n0 upgraded, 0 newly installed, 0 to remove and 7 not upgraded.\n", "output_type": "stream"}]}, {"cell_type": "markdown", "source": "Mapping CCS reads", "metadata": {}}, {"cell_type": "code", "source": "%%bash\nmkdir -p cram\n\nminimap2 -t 32 -R '@RG\\tID:1\\tSM:HG002' -ax asm20 ref/hs37d5.fa.gz \\\nreads/HG002.Sequel.15kb.pbmm2.hs37d5.whatshap.haplotag.RTG.10x.trio.MHConly.fastq.gz | \\\nsamtools view -T ref/hs37d5.fa.gz -C - > cram/HG002.PacBio.15kbCCS.cram", "metadata": {"trusted": true}, "execution_count": 8, "outputs": [{"name": "stderr", "text": "[samfaipath] build FASTA index...\n[M::mm_idx_gen::74.091*2.10] collected minimizers\n[M::mm_idx_gen::88.173*3.02] sorted minimizers\n[M::main::88.173*3.02] loaded/built the index for 86 target sequence(s)\n[M::mm_mapopt_update::97.678*2.82] mid_occ = 127\n[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 0; #seq: 86\n[M::mm_idx_stat::103.957*2.71] distinct minimizers: 388792771 (92.02% are singletons); average occurrences: 1.359; average spacing: 5.936\n[M::worker_pipeline::185.796*3.27] mapped 25922 sequences\n[M::main] Version: 2.17-r941\n[M::main] CMD: minimap2 -t 32 -R @RG\\tID:1\\tSM:HG002 -ax asm20 ref/hs37d5.fa.gz reads/HG002.Sequel.15kb.pbmm2.hs37d5.whatshap.haplotag.RTG.10x.trio.MHConly.fastq.gz\n[M::main] Real time: 185.971 sec; CPU: 607.652 sec; Peak RSS: 14.109 GB\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "%%bash\nsamtools sort -@ 32 -O cram -o cram/HG002.PacBio.15kbCCS.sorted.cram cram/HG002.PacBio.15kbCCS.cram\nsamtools index cram/HG002.PacBio.15kbCCS.sorted.cram", "metadata": {"trusted": true}, "execution_count": 9, "outputs": [{"name": "stderr", "text": "[W::cram_populate_ref] Creating reference cache directory /home/dnanexus/.cache/hts-ref\nThis may become large; see the samtools(1) manual page REF_CACHE discussion\n[bam_sort_core] merging from 0 files and 32 in-memory blocks...\n", "output_type": "stream"}]}, {"cell_type": "markdown", "source": "Mapping ONT reads", "metadata": {}}, {"cell_type": "code", "source": "%%bash\nminimap2 -t 32 -R '@RG\\tID:1\\tSM:HG002' -ax map-ont ref/hs37d5.fa.gz \\\nreads/HG002_ultra-long-ont_hs37d5_phased_reheader.MHConly.fastq.gz | \\\nsamtools view -T ref/hs37d5.fa.gz -C - > cram/HG002.ultra-long-ont_hs37d5_phased_reheader.cram", "metadata": {"trusted": true}, "execution_count": 10, "outputs": [{"name": "stderr", "text": "[M::mm_idx_gen::67.024*2.21] collected minimizers\n[M::mm_idx_gen::82.089*3.28] sorted minimizers\n[M::main::82.090*3.28] loaded/built the index for 86 target sequence(s)\n[M::mm_mapopt_update::84.435*3.21] mid_occ = 617\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 86\n[M::mm_idx_stat::86.634*3.16] distinct minimizers: 100235800 (38.76% are singletons); average occurrences: 5.452; average spacing: 5.741\n[M::worker_pipeline::219.079*4.38] mapped 15421 sequences\n[M::main] Version: 2.17-r941\n[M::main] CMD: minimap2 -t 32 -R @RG\\tID:1\\tSM:HG002 -ax map-ont ref/hs37d5.fa.gz reads/HG002_ultra-long-ont_hs37d5_phased_reheader.MHConly.fastq.gz\n[M::main] Real time: 219.301 sec; CPU: 959.692 sec; Peak RSS: 20.795 GB\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "%%bash\nsamtools sort  -@ 32 -O cram -o cram/HG002.ultra-long-ont_hs37d5_phased_reheader.sorted.cram cram/HG002.ultra-long-ont_hs37d5_phased_reheader.cram \nsamtools index cram/HG002.ultra-long-ont_hs37d5_phased_reheader.sorted.cram", "metadata": {"trusted": true}, "execution_count": 11, "outputs": [{"name": "stderr", "text": "[bam_sort_core] merging from 0 files and 32 in-memory blocks...\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "%%bash\ndx download -f /HG002_reanalysis/pacbio-15kb-hapsort-wgs.vcf.gz\nmkdir -p vcf/deepvariant/\nmv pacbio-15kb-hapsort-wgs.vcf.gz vcf/deepvariant/", "metadata": {"trusted": true}, "execution_count": 13, "outputs": []}, {"cell_type": "code", "source": "%%bash\ntabix -p vcf vcf/deepvariant/pacbio-15kb-hapsort-wgs.vcf.gz", "metadata": {"trusted": true}, "execution_count": 14, "outputs": []}, {"cell_type": "markdown", "source": "Select het SNVs from DeepVariant call set. The goal is to use this as a starting point to construct a super confident set of heterozygous SNVs that we can use to partition reads", "metadata": {}}, {"cell_type": "code", "source": "%%bash\nbcftools view -g het -v snps vcf/deepvariant/pacbio-15kb-hapsort-wgs.vcf.gz 6:28477797-33448354 > vcf/deepvariant/mhc.het.snps.vcf", "metadata": {"trusted": true}, "execution_count": 15, "outputs": []}, {"cell_type": "markdown", "source": "Re-genotype these SNV from ONT reads", "metadata": {}}, {"cell_type": "code", "source": "%%bash\nzcat ref/hs37d5.fa.gz > ref/hs37d5.fa", "metadata": {"trusted": true}, "execution_count": 16, "outputs": []}, {"cell_type": "code", "source": "%%bash\nmkdir -p whatshap/genotype/\nwhatshap genotype --ignore-read-groups --chromosome 6 \\\n--reference ref/hs37d5.fa \\\n-o whatshap/genotype/HG002.ultra-long-ont.vcf \\\nvcf/deepvariant/mhc.het.snps.vcf \\\ncram/HG002.ultra-long-ont_hs37d5_phased_reheader.sorted.cram > whatshap/genotype/HG002.ultra-long-ont.log 2>&1", "metadata": {"trusted": true}, "execution_count": 17, "outputs": []}, {"cell_type": "markdown", "source": "Re-genotype these SNV from CCS reads", "metadata": {}}, {"cell_type": "code", "source": "%%bash\nwhatshap genotype --ignore-read-groups --chromosome 6 \\\n--reference ref/hs37d5.fa \\\n-o whatshap/genotype/HG002.PacBio.15kbCCS.vcf \\\nvcf/deepvariant/mhc.het.snps.vcf \\\ncram/HG002.PacBio.15kbCCS.sorted.cram > whatshap/genotype/HG002.PacBio.15kbCCS.log 2>&1", "metadata": {"trusted": true}, "execution_count": 18, "outputs": []}, {"cell_type": "markdown", "source": "Find the intersection, i.e. only retain those that are het accoring to DV/CCS, WhatHap/ONT, and WhatsHap/CCS", "metadata": {}}, {"cell_type": "code", "source": "%%bash\nbcftools view -g het whatshap/genotype/HG002.ultra-long-ont.vcf > whatshap/genotype/HG002.ultra-long-ont.hetonly.vcf\nbgzip whatshap/genotype/HG002.ultra-long-ont.hetonly.vcf\nrm HG002.ultra-long-ont.hetonly.vcf.gz.tbi\ntabix -p vcf -f whatshap/genotype/HG002.ultra-long-ont.hetonly.vcf.gz\n\nbcftools view -g het whatshap/genotype/HG002.PacBio.15kbCCS.vcf > whatshap/genotype/HG002.PacBio.15kbCCS.hetonly.vcf\nbgzip whatshap/genotype/HG002.PacBio.15kbCCS.hetonly.vcf\nrm HG002.PacBio.15kbCCS.hetonly.vcf.gz.tbi\ntabix -p vcf -f whatshap/genotype/HG002.PacBio.15kbCCS.hetonly.vcf.gz\n\nmkdir -p whatshap/genotype-intersect\nbcftools isec -p whatshap/genotype-intersect whatshap/genotype/HG002.ultra-long-ont.hetonly.vcf.gz whatshap/genotype/HG002.PacBio.15kbCCS.hetonly.vcf.gz \ncp whatshap/genotype-intersect/0003.vcf whatshap/confident-hets.vcf", "metadata": {"trusted": true}, "execution_count": 19, "outputs": [{"name": "stderr", "text": "rm: cannot remove 'HG002.ultra-long-ont.hetonly.vcf.gz.tbi': No such file or directory\nrm: cannot remove 'HG002.PacBio.15kbCCS.hetonly.vcf.gz.tbi': No such file or directory\n", "output_type": "stream"}]}, {"cell_type": "markdown", "source": "Extract 10x phased blocks produced by LongRanger ", "metadata": {}}, {"cell_type": "code", "source": "%%bash\nmkdir -p 10x\ncp NA24385.GRCh37.phased_variants.vcf.gz 10x/\ncd 10x\nrm NA24385.GRCh37.phased_variants.vcf.gz.tbi\ntabix -p vcf -f NA24385.GRCh37.phased_variants.vcf.gz\nbcftools view NA24385.GRCh37.phased_variants.vcf.gz chr6:28477797-33448354 | \\\n    awk 'BEGIN {OFS=\"\\t\"} $1 == \"chr6\" {$1=\"6\"} $1==\"#CHROM\" {$10=\"HG002\"} {print}'| \\\n    bgzip > NA24385.GRCh37.phased_variants.mhc.vcf.gz", "metadata": {"trusted": true}, "execution_count": 20, "outputs": [{"name": "stderr", "text": "rm: cannot remove 'NA24385.GRCh37.phased_variants.vcf.gz.tbi': No such file or directory\n", "output_type": "stream"}]}, {"cell_type": "markdown", "source": "Now run phasing from multiple combinations of data sources", "metadata": {}}, {"cell_type": "code", "source": "%%bash\nrm -rf whatshap/phase/\nmkdir -p whatshap/phase/\n\nwhatshap phase --ignore-read-groups --chromosome 6 \\\n--reference ref/hs37d5.fa \\\n-o whatshap/phase/HG002.PacBio.15kbCCS.vcf \\\nwhatshap/confident-hets.vcf \\\ncram/HG002.PacBio.15kbCCS.sorted.cram > whatshap/phase/HG002.PacBio.15kbCCS.log 2>&1\n\nbgzip -f whatshap/phase/HG002.PacBio.15kbCCS.vcf\ntabix -p vcf -f whatshap/phase/HG002.PacBio.15kbCCS.vcf.gz\n\nwhatshap phase --ignore-read-groups --chromosome 6 \\\n--reference ref/hs37d5.fa -o whatshap/phase/HG002.ultra-long-ont.vcf \\\nwhatshap/confident-hets.vcf \\\ncram/HG002.ultra-long-ont_hs37d5_phased_reheader.sorted.cram > whatshap/phase/HG002.ultra-long-ont.log 2>&1\n\nbgzip -f whatshap/phase/HG002.ultra-long-ont.vcf\ntabix -p vcf -f  whatshap/phase/HG002.ultra-long-ont.vcf.gz\n\nwhatshap phase --ignore-read-groups --chromosome 6 \\\n--reference ref/hs37d5.fa \\\n-o whatshap/phase/HG002.PacBio+10x.vcf \\\nwhatshap/confident-hets.vcf cram/HG002.PacBio.15kbCCS.sorted.cram \\\n10x/NA24385.GRCh37.phased_variants.mhc.vcf.gz > whatshap/phase/HG002.PacBio+10x.log 2>&1\n\nbgzip -f whatshap/phase/HG002.PacBio+10x.vcf\ntabix -p vcf -f whatshap/phase/HG002.PacBio+10x.vcf.gz\n\n", "metadata": {"trusted": true}, "execution_count": 21, "outputs": []}, {"cell_type": "code", "source": "%%bash\n\n\nwhatshap phase --chromosome 6 \\\n--reference ref/hs37d5.fa \\\n-o whatshap/phase/HG002.ultra-long-ont+10x.vcf \\\nwhatshap/confident-hets.vcf \\\ncram/HG002.ultra-long-ont_hs37d5_phased_reheader.sorted.cram \\\n10x/NA24385.GRCh37.phased_variants.mhc.vcf.gz > whatshap/phase/HG002.ultra-long-ont+10x.log 2>&1\n\nbgzip -f whatshap/phase/HG002.ultra-long-ont+10x.vcf\ntabix -p vcf -f whatshap/phase/HG002.ultra-long-ont+10x.vcf.gz\n\n\nwhatshap haplotag --ignore-read-groups \\\n--output-haplotag-list cram/HG002.PacBio.15kbCCS.sorted.tags-newsplit.tsv \\\n--reference ref/hs37d5.fa \\\n-o cram/HG002.PacBio.15kbCCS.sorted.tagged-newsplit.cram \\\nwhatshap/phase/HG002.ultra-long-ont+10x.vcf.gz \\\ncram/HG002.PacBio.15kbCCS.sorted.cram\n\nwhatshap haplotag --ignore-read-groups \\\n--output-haplotag-list cram/HG002.ultra-long-ont.sorted.tags-newsplit.tsv \\\n--reference ref/hs37d5.fa \\\n-o cram/HG002.ultra-long-ont.sorted.tagged-newsplit.cram \\\nwhatshap/phase/HG002.ultra-long-ont+10x.vcf.gz cram/HG002.ultra-long-ont_hs37d5_phased_reheader.sorted.cram \n", "metadata": {"trusted": true}, "execution_count": 30, "outputs": [{"name": "stderr", "text": "Found 1 samples in VCF file\nSamples in BAM file: HG002\nProcessing alignments on chromosome 1\n... found 0 variants for chromosome 1 in VCF\nDetecting alleles in reads mapped to chromosome 1 for sample None ...\nProcessing alignments on chromosome 2\n... found 0 variants for chromosome 2 in VCF\nDetecting alleles in reads mapped to chromosome 2 for sample None ...\nProcessing alignments on chromosome 3\n... found 0 variants for chromosome 3 in VCF\nDetecting alleles in reads mapped to chromosome 3 for sample None ...\nProcessing alignments on chromosome 4\n... found 0 variants for chromosome 4 in VCF\nDetecting alleles in reads mapped to chromosome 4 for sample None ...\nProcessing alignments on chromosome 5\n... found 0 variants for chromosome 5 in VCF\nDetecting alleles in reads mapped to chromosome 5 for sample None ...\nProcessing alignments on chromosome 6\n... found 12283 variants for chromosome 6 in VCF\nDetecting alleles in reads mapped to chromosome 6 for sample None ...\nProcessing alignments on chromosome 7\n... found 0 variants for chromosome 7 in VCF\nDetecting alleles in reads mapped to chromosome 7 for sample None ...\nProcessing alignments on chromosome 8\n... found 0 variants for chromosome 8 in VCF\nDetecting alleles in reads mapped to chromosome 8 for sample None ...\nProcessing alignments on chromosome 9\n... found 0 variants for chromosome 9 in VCF\nDetecting alleles in reads mapped to chromosome 9 for sample None ...\nProcessing alignments on chromosome 10\n... found 0 variants for chromosome 10 in VCF\nDetecting alleles in reads mapped to chromosome 10 for sample None ...\nProcessing alignments on chromosome 11\n... found 0 variants for chromosome 11 in VCF\nDetecting alleles in reads mapped to chromosome 11 for sample None ...\nProcessing alignments on chromosome 12\n... found 0 variants for chromosome 12 in VCF\nDetecting alleles in reads mapped to chromosome 12 for sample None ...\nProcessing alignments on chromosome 13\n... found 0 variants for chromosome 13 in VCF\nDetecting alleles in reads mapped to chromosome 13 for sample None ...\nProcessing alignments on chromosome 14\n... found 0 variants for chromosome 14 in VCF\nDetecting alleles in reads mapped to chromosome 14 for sample None ...\nProcessing alignments on chromosome 15\n... found 0 variants for chromosome 15 in VCF\nDetecting alleles in reads mapped to chromosome 15 for sample None ...\nProcessing alignments on chromosome 16\n... found 0 variants for chromosome 16 in VCF\nDetecting alleles in reads mapped to chromosome 16 for sample None ...\nProcessing alignments on chromosome 17\n... found 0 variants for chromosome 17 in VCF\nDetecting alleles in reads mapped to chromosome 17 for sample None ...\nProcessing alignments on chromosome 18\n... found 0 variants for chromosome 18 in VCF\nDetecting alleles in reads mapped to chromosome 18 for sample None ...\nProcessing alignments on chromosome 19\n... found 0 variants for chromosome 19 in VCF\nDetecting alleles in reads mapped to chromosome 19 for sample None ...\nProcessing alignments on chromosome 20\n... found 0 variants for chromosome 20 in VCF\nDetecting alleles in reads mapped to chromosome 20 for sample None ...\nProcessing alignments on chromosome 21\n... found 0 variants for chromosome 21 in VCF\nDetecting alleles in reads mapped to chromosome 21 for sample None ...\nProcessing alignments on chromosome 22\n... found 0 variants for chromosome 22 in VCF\nDetecting alleles in reads mapped to chromosome 22 for sample None ...\nProcessing alignments on chromosome X\n... found 0 variants for chromosome X in VCF\nDetecting alleles in reads mapped to chromosome X for sample None ...\nProcessing alignments on chromosome Y\n... found 0 variants for chromosome Y in VCF\nDetecting alleles in reads mapped to chromosome Y for sample None ...\nProcessing alignments on chromosome GL000208.1\n... found 0 variants for chromosome GL000208.1 in VCF\nDetecting alleles in reads mapped to chromosome GL000208.1 for sample None ...\nProcessing alignments on chromosome GL000191.1\n... found 0 variants for chromosome GL000191.1 in VCF\nDetecting alleles in reads mapped to chromosome GL000191.1 for sample None ...\nProcessing alignments on chromosome GL000220.1\n... found 0 variants for chromosome GL000220.1 in VCF\nDetecting alleles in reads mapped to chromosome GL000220.1 for sample None ...\nProcessing alignments on chromosome GL000216.1\n... found 0 variants for chromosome GL000216.1 in VCF\nDetecting alleles in reads mapped to chromosome GL000216.1 for sample None ...\nProcessing alignments on chromosome GL000224.1\n... found 0 variants for chromosome GL000224.1 in VCF\nDetecting alleles in reads mapped to chromosome GL000224.1 for sample None ...\nProcessing alignments on chromosome GL000223.1\n... found 0 variants for chromosome GL000223.1 in VCF\nDetecting alleles in reads mapped to chromosome GL000223.1 for sample None ...\nProcessing alignments on chromosome GL000222.1\n... found 0 variants for chromosome GL000222.1 in VCF\nDetecting alleles in reads mapped to chromosome GL000222.1 for sample None ...\nProcessing alignments on chromosome GL000225.1\n... found 0 variants for chromosome GL000225.1 in VCF\nDetecting alleles in reads mapped to chromosome GL000225.1 for sample None ...\nProcessing alignments on chromosome hs37d5\n... found 0 variants for chromosome hs37d5 in VCF\nDetecting alleles in reads mapped to chromosome hs37d5 for sample None ...\n\n== SUMMARY ==\nTotal alignments processed:                     33490\nAlignments that could be tagged:                22389\nAlignments spanning multiple phase sets:            0\nFound 1 samples in VCF file\nSamples in BAM file: HG002\nProcessing alignments on chromosome 1\n... found 0 variants for chromosome 1 in VCF\nDetecting alleles in reads mapped to chromosome 1 for sample None ...\nProcessing alignments on chromosome 2\n... found 0 variants for chromosome 2 in VCF\nDetecting alleles in reads mapped to chromosome 2 for sample None ...\nProcessing alignments on chromosome 3\n... found 0 variants for chromosome 3 in VCF\nDetecting alleles in reads mapped to chromosome 3 for sample None ...\nProcessing alignments on chromosome 4\n... found 0 variants for chromosome 4 in VCF\nDetecting alleles in reads mapped to chromosome 4 for sample None ...\nProcessing alignments on chromosome 5\n... found 0 variants for chromosome 5 in VCF\nDetecting alleles in reads mapped to chromosome 5 for sample None ...\nProcessing alignments on chromosome 6\n... found 12283 variants for chromosome 6 in VCF\nDetecting alleles in reads mapped to chromosome 6 for sample None ...\nProcessing alignments on chromosome 7\n... found 0 variants for chromosome 7 in VCF\nDetecting alleles in reads mapped to chromosome 7 for sample None ...\nProcessing alignments on chromosome 8\n... found 0 variants for chromosome 8 in VCF\nDetecting alleles in reads mapped to chromosome 8 for sample None ...\nProcessing alignments on chromosome 9\n... found 0 variants for chromosome 9 in VCF\nDetecting alleles in reads mapped to chromosome 9 for sample None ...\nProcessing alignments on chromosome 10\n... found 0 variants for chromosome 10 in VCF\nDetecting alleles in reads mapped to chromosome 10 for sample None ...\nProcessing alignments on chromosome 11\n... found 0 variants for chromosome 11 in VCF\nDetecting alleles in reads mapped to chromosome 11 for sample None ...\nProcessing alignments on chromosome 12\n... found 0 variants for chromosome 12 in VCF\nDetecting alleles in reads mapped to chromosome 12 for sample None ...\nProcessing alignments on chromosome 13\n... found 0 variants for chromosome 13 in VCF\nDetecting alleles in reads mapped to chromosome 13 for sample None ...\nProcessing alignments on chromosome 14\n... found 0 variants for chromosome 14 in VCF\nDetecting alleles in reads mapped to chromosome 14 for sample None ...\nProcessing alignments on chromosome 15\n... found 0 variants for chromosome 15 in VCF\nDetecting alleles in reads mapped to chromosome 15 for sample None ...\nProcessing alignments on chromosome 16\n... found 0 variants for chromosome 16 in VCF\nDetecting alleles in reads mapped to chromosome 16 for sample None ...\nProcessing alignments on chromosome 17\n... found 0 variants for chromosome 17 in VCF\nDetecting alleles in reads mapped to chromosome 17 for sample None ...\nProcessing alignments on chromosome 18\n... found 0 variants for chromosome 18 in VCF\nDetecting alleles in reads mapped to chromosome 18 for sample None ...\nProcessing alignments on chromosome 19\n... found 0 variants for chromosome 19 in VCF\nDetecting alleles in reads mapped to chromosome 19 for sample None ...\nProcessing alignments on chromosome 20\n... found 0 variants for chromosome 20 in VCF\nDetecting alleles in reads mapped to chromosome 20 for sample None ...\nProcessing alignments on chromosome 21\n... found 0 variants for chromosome 21 in VCF\nDetecting alleles in reads mapped to chromosome 21 for sample None ...\nProcessing alignments on chromosome 22\n... found 0 variants for chromosome 22 in VCF\nDetecting alleles in reads mapped to chromosome 22 for sample None ...\nProcessing alignments on chromosome X\n... found 0 variants for chromosome X in VCF\nDetecting alleles in reads mapped to chromosome X for sample None ...\nProcessing alignments on chromosome Y\n... found 0 variants for chromosome Y in VCF\nDetecting alleles in reads mapped to chromosome Y for sample None ...\nProcessing alignments on chromosome GL000247.1\n... found 0 variants for chromosome GL000247.1 in VCF\nDetecting alleles in reads mapped to chromosome GL000247.1 for sample None ...\nProcessing alignments on chromosome GL000191.1\n... found 0 variants for chromosome GL000191.1 in VCF\nDetecting alleles in reads mapped to chromosome GL000191.1 for sample None ...\nProcessing alignments on chromosome GL000220.1\n... found 0 variants for chromosome GL000220.1 in VCF\nDetecting alleles in reads mapped to chromosome GL000220.1 for sample None ...\nProcessing alignments on chromosome GL000217.1\n... found 0 variants for chromosome GL000217.1 in VCF\nDetecting alleles in reads mapped to chromosome GL000217.1 for sample None ...\nProcessing alignments on chromosome GL000216.1\n... found 0 variants for chromosome GL000216.1 in VCF\nDetecting alleles in reads mapped to chromosome GL000216.1 for sample None ...\nProcessing alignments on chromosome GL000224.1\n... found 0 variants for chromosome GL000224.1 in VCF\nDetecting alleles in reads mapped to chromosome GL000224.1 for sample None ...\nProcessing alignments on chromosome GL000223.1\n... found 0 variants for chromosome GL000223.1 in VCF\nDetecting alleles in reads mapped to chromosome GL000223.1 for sample None ...\nProcessing alignments on chromosome hs37d5\n... found 0 variants for chromosome hs37d5 in VCF\nDetecting alleles in reads mapped to chromosome hs37d5 for sample None ...\n\n== SUMMARY ==\nTotal alignments processed:                     20737\nAlignments that could be tagged:                10778\nAlignments spanning multiple phase sets:            0\n", "output_type": "stream"}]}, {"cell_type": "markdown", "source": "Tag reads and split FASTQs ", "metadata": {}}, {"cell_type": "code", "source": "%%bash\n\nwhatshap haplotag --ignore-read-groups \\\n--output-haplotag-list cram/HG002.PacBio.15kbCCS.sorted.tags.tsv \\\n--reference ref/hs37d5.fa \\\n-o cram/HG002.PacBio.15kbCCS.sorted.tagged.cram \\\nwhatshap/phase/HG002.PacBio+10x.vcf.gz \\\ncram/HG002.PacBio.15kbCCS.sorted.cram\n\nwhatshap haplotag --ignore-read-groups --output-haplotag-list \\\ncram/HG002.ultra-long-ont_hs37d5_phased_reheader.sorted.tags.tsv \\\n--reference ref/hs37d5.fa \\\n-o cram/HG002_ultra-long-ont_hs37d5_phased_reheader.sorted.tagged.cram \\\nwhatshap/phase/HG002.PacBio+10x.vcf.gz \\\ncram/HG002.ultra-long-ont_hs37d5_phased_reheader.sorted.cram\n", "metadata": {"trusted": true}, "execution_count": 22, "outputs": [{"name": "stderr", "text": "Found 1 samples in VCF file\nSamples in BAM file: HG002\nProcessing alignments on chromosome 1\n... found 0 variants for chromosome 1 in VCF\nDetecting alleles in reads mapped to chromosome 1 for sample None ...\nProcessing alignments on chromosome 2\n... found 0 variants for chromosome 2 in VCF\nDetecting alleles in reads mapped to chromosome 2 for sample None ...\nProcessing alignments on chromosome 3\n... found 0 variants for chromosome 3 in VCF\nDetecting alleles in reads mapped to chromosome 3 for sample None ...\nProcessing alignments on chromosome 4\n... found 0 variants for chromosome 4 in VCF\nDetecting alleles in reads mapped to chromosome 4 for sample None ...\nProcessing alignments on chromosome 5\n... found 0 variants for chromosome 5 in VCF\nDetecting alleles in reads mapped to chromosome 5 for sample None ...\nProcessing alignments on chromosome 6\n... found 12283 variants for chromosome 6 in VCF\nDetecting alleles in reads mapped to chromosome 6 for sample None ...\nProcessing alignments on chromosome 7\n... found 0 variants for chromosome 7 in VCF\nDetecting alleles in reads mapped to chromosome 7 for sample None ...\nProcessing alignments on chromosome 8\n... found 0 variants for chromosome 8 in VCF\nDetecting alleles in reads mapped to chromosome 8 for sample None ...\nProcessing alignments on chromosome 9\n... found 0 variants for chromosome 9 in VCF\nDetecting alleles in reads mapped to chromosome 9 for sample None ...\nProcessing alignments on chromosome 10\n... found 0 variants for chromosome 10 in VCF\nDetecting alleles in reads mapped to chromosome 10 for sample None ...\nProcessing alignments on chromosome 11\n... found 0 variants for chromosome 11 in VCF\nDetecting alleles in reads mapped to chromosome 11 for sample None ...\nProcessing alignments on chromosome 12\n... found 0 variants for chromosome 12 in VCF\nDetecting alleles in reads mapped to chromosome 12 for sample None ...\nProcessing alignments on chromosome 13\n... found 0 variants for chromosome 13 in VCF\nDetecting alleles in reads mapped to chromosome 13 for sample None ...\nProcessing alignments on chromosome 14\n... found 0 variants for chromosome 14 in VCF\nDetecting alleles in reads mapped to chromosome 14 for sample None ...\nProcessing alignments on chromosome 15\n... found 0 variants for chromosome 15 in VCF\nDetecting alleles in reads mapped to chromosome 15 for sample None ...\nProcessing alignments on chromosome 16\n... found 0 variants for chromosome 16 in VCF\nDetecting alleles in reads mapped to chromosome 16 for sample None ...\nProcessing alignments on chromosome 17\n... found 0 variants for chromosome 17 in VCF\nDetecting alleles in reads mapped to chromosome 17 for sample None ...\nProcessing alignments on chromosome 18\n... found 0 variants for chromosome 18 in VCF\nDetecting alleles in reads mapped to chromosome 18 for sample None ...\nProcessing alignments on chromosome 19\n... found 0 variants for chromosome 19 in VCF\nDetecting alleles in reads mapped to chromosome 19 for sample None ...\nProcessing alignments on chromosome 20\n... found 0 variants for chromosome 20 in VCF\nDetecting alleles in reads mapped to chromosome 20 for sample None ...\nProcessing alignments on chromosome 21\n... found 0 variants for chromosome 21 in VCF\nDetecting alleles in reads mapped to chromosome 21 for sample None ...\nProcessing alignments on chromosome 22\n... found 0 variants for chromosome 22 in VCF\nDetecting alleles in reads mapped to chromosome 22 for sample None ...\nProcessing alignments on chromosome X\n... found 0 variants for chromosome X in VCF\nDetecting alleles in reads mapped to chromosome X for sample None ...\nProcessing alignments on chromosome Y\n... found 0 variants for chromosome Y in VCF\nDetecting alleles in reads mapped to chromosome Y for sample None ...\nProcessing alignments on chromosome GL000208.1\n... found 0 variants for chromosome GL000208.1 in VCF\nDetecting alleles in reads mapped to chromosome GL000208.1 for sample None ...\nProcessing alignments on chromosome GL000191.1\n... found 0 variants for chromosome GL000191.1 in VCF\nDetecting alleles in reads mapped to chromosome GL000191.1 for sample None ...\nProcessing alignments on chromosome GL000220.1\n... found 0 variants for chromosome GL000220.1 in VCF\nDetecting alleles in reads mapped to chromosome GL000220.1 for sample None ...\nProcessing alignments on chromosome GL000216.1\n... found 0 variants for chromosome GL000216.1 in VCF\nDetecting alleles in reads mapped to chromosome GL000216.1 for sample None ...\nProcessing alignments on chromosome GL000224.1\n... found 0 variants for chromosome GL000224.1 in VCF\nDetecting alleles in reads mapped to chromosome GL000224.1 for sample None ...\nProcessing alignments on chromosome GL000223.1\n... found 0 variants for chromosome GL000223.1 in VCF\nDetecting alleles in reads mapped to chromosome GL000223.1 for sample None ...\nProcessing alignments on chromosome GL000222.1\n... found 0 variants for chromosome GL000222.1 in VCF\nDetecting alleles in reads mapped to chromosome GL000222.1 for sample None ...\nProcessing alignments on chromosome GL000225.1\n... found 0 variants for chromosome GL000225.1 in VCF\nDetecting alleles in reads mapped to chromosome GL000225.1 for sample None ...\nProcessing alignments on chromosome hs37d5\n... found 0 variants for chromosome hs37d5 in VCF\nDetecting alleles in reads mapped to chromosome hs37d5 for sample None ...\n\n== SUMMARY ==\nTotal alignments processed:                     33490\nAlignments that could be tagged:                22335\nAlignments spanning multiple phase sets:            0\nFound 1 samples in VCF file\nSamples in BAM file: HG002\nProcessing alignments on chromosome 1\n... found 0 variants for chromosome 1 in VCF\nDetecting alleles in reads mapped to chromosome 1 for sample None ...\nProcessing alignments on chromosome 2\n... found 0 variants for chromosome 2 in VCF\nDetecting alleles in reads mapped to chromosome 2 for sample None ...\nProcessing alignments on chromosome 3\n... found 0 variants for chromosome 3 in VCF\nDetecting alleles in reads mapped to chromosome 3 for sample None ...\nProcessing alignments on chromosome 4\n... found 0 variants for chromosome 4 in VCF\nDetecting alleles in reads mapped to chromosome 4 for sample None ...\nProcessing alignments on chromosome 5\n... found 0 variants for chromosome 5 in VCF\nDetecting alleles in reads mapped to chromosome 5 for sample None ...\nProcessing alignments on chromosome 6\n... found 12283 variants for chromosome 6 in VCF\nDetecting alleles in reads mapped to chromosome 6 for sample None ...\nProcessing alignments on chromosome 7\n... found 0 variants for chromosome 7 in VCF\nDetecting alleles in reads mapped to chromosome 7 for sample None ...\nProcessing alignments on chromosome 8\n... found 0 variants for chromosome 8 in VCF\nDetecting alleles in reads mapped to chromosome 8 for sample None ...\nProcessing alignments on chromosome 9\n... found 0 variants for chromosome 9 in VCF\nDetecting alleles in reads mapped to chromosome 9 for sample None ...\nProcessing alignments on chromosome 10\n... found 0 variants for chromosome 10 in VCF\nDetecting alleles in reads mapped to chromosome 10 for sample None ...\nProcessing alignments on chromosome 11\n... found 0 variants for chromosome 11 in VCF\nDetecting alleles in reads mapped to chromosome 11 for sample None ...\nProcessing alignments on chromosome 12\n... found 0 variants for chromosome 12 in VCF\nDetecting alleles in reads mapped to chromosome 12 for sample None ...\nProcessing alignments on chromosome 13\n... found 0 variants for chromosome 13 in VCF\nDetecting alleles in reads mapped to chromosome 13 for sample None ...\nProcessing alignments on chromosome 14\n... found 0 variants for chromosome 14 in VCF\nDetecting alleles in reads mapped to chromosome 14 for sample None ...\nProcessing alignments on chromosome 15\n... found 0 variants for chromosome 15 in VCF\nDetecting alleles in reads mapped to chromosome 15 for sample None ...\nProcessing alignments on chromosome 16\n... found 0 variants for chromosome 16 in VCF\nDetecting alleles in reads mapped to chromosome 16 for sample None ...\nProcessing alignments on chromosome 17\n... found 0 variants for chromosome 17 in VCF\nDetecting alleles in reads mapped to chromosome 17 for sample None ...\nProcessing alignments on chromosome 18\n... found 0 variants for chromosome 18 in VCF\nDetecting alleles in reads mapped to chromosome 18 for sample None ...\nProcessing alignments on chromosome 19\n... found 0 variants for chromosome 19 in VCF\nDetecting alleles in reads mapped to chromosome 19 for sample None ...\nProcessing alignments on chromosome 20\n... found 0 variants for chromosome 20 in VCF\nDetecting alleles in reads mapped to chromosome 20 for sample None ...\nProcessing alignments on chromosome 21\n... found 0 variants for chromosome 21 in VCF\nDetecting alleles in reads mapped to chromosome 21 for sample None ...\nProcessing alignments on chromosome 22\n... found 0 variants for chromosome 22 in VCF\nDetecting alleles in reads mapped to chromosome 22 for sample None ...\nProcessing alignments on chromosome X\n... found 0 variants for chromosome X in VCF\nDetecting alleles in reads mapped to chromosome X for sample None ...\nProcessing alignments on chromosome Y\n... found 0 variants for chromosome Y in VCF\nDetecting alleles in reads mapped to chromosome Y for sample None ...\nProcessing alignments on chromosome GL000247.1\n... found 0 variants for chromosome GL000247.1 in VCF\nDetecting alleles in reads mapped to chromosome GL000247.1 for sample None ...\nProcessing alignments on chromosome GL000191.1\n... found 0 variants for chromosome GL000191.1 in VCF\nDetecting alleles in reads mapped to chromosome GL000191.1 for sample None ...\nProcessing alignments on chromosome GL000220.1\n... found 0 variants for chromosome GL000220.1 in VCF\nDetecting alleles in reads mapped to chromosome GL000220.1 for sample None ...\nProcessing alignments on chromosome GL000217.1\n... found 0 variants for chromosome GL000217.1 in VCF\nDetecting alleles in reads mapped to chromosome GL000217.1 for sample None ...\nProcessing alignments on chromosome GL000216.1\n... found 0 variants for chromosome GL000216.1 in VCF\nDetecting alleles in reads mapped to chromosome GL000216.1 for sample None ...\nProcessing alignments on chromosome GL000224.1\n... found 0 variants for chromosome GL000224.1 in VCF\nDetecting alleles in reads mapped to chromosome GL000224.1 for sample None ...\nProcessing alignments on chromosome GL000223.1\n... found 0 variants for chromosome GL000223.1 in VCF\nDetecting alleles in reads mapped to chromosome GL000223.1 for sample None ...\nProcessing alignments on chromosome hs37d5\n... found 0 variants for chromosome hs37d5 in VCF\nDetecting alleles in reads mapped to chromosome hs37d5 for sample None ...\n\n== SUMMARY ==\nTotal alignments processed:                     20737\nAlignments that could be tagged:                10767\nAlignments spanning multiple phase sets:           23\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "%%writefile split_fastq_w_untagged.sh \n#!/bin/bash\n\nif [ \"$#\" -ne 2 ]; then\n    echo \"Usage: $0 <input.fastq.gz> <taglist.tsv>\"\n    exit 1\nfi\n\nfastq=$1\nlistfile=$2\n\n\nfor block in $(tail -n +2 $listfile |cut -f 3|grep -v 'none' | sort |uniq) ; do\n        echo \"Block $block\"\n        whatshap split --output-h1 ${fastq%.fastq.gz}.$block.H1.fastq.gz \\\n                       --output-h2 ${fastq%.fastq.gz}.$block.H2.fastq.gz \\\n                       --output-untagged ${fastq%.fastq.gz}.$block.untagged.fastq.gz $fastq <(awk \"\\$3==$block\" $listfile)\ndone", "metadata": {"trusted": true}, "execution_count": 24, "outputs": [{"name": "stdout", "text": "Writing split_fastq_w_untagged.sh\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "%%bash\nbash ./split_fastq_w_untagged.sh \\\nreads/HG002.Sequel.15kb.pbmm2.hs37d5.whatshap.haplotag.RTG.10x.trio.MHConly.fastq.gz \\\ncram/HG002.PacBio.15kbCCS.sorted.tags.tsv\n\nbash ./split_fastq_w_untagged.sh \\\nreads/HG002_ultra-long-ont_hs37d5_phased_reheader.MHConly.fastq.gz \\\ncram/HG002.ultra-long-ont_hs37d5_phased_reheader.sorted.tags.tsv", "metadata": {"trusted": true}, "execution_count": 25, "outputs": [{"name": "stdout", "text": "Block 28498559\nBlock 31869673\nBlock 31968164\nBlock 32457470\nBlock 28498559\nBlock 31869673\nBlock 31968164\nBlock 32457470\n", "output_type": "stream"}, {"name": "stderr", "text": "Reading /dev/fd/63\n... read 15270 records:\n...   15270 reads are assigned to a haplotype.\n\n== SUMMARY ==\nTotal reads processed:                     25922\nReading /dev/fd/63\n... read 93 records:\n...   93 reads are assigned to a haplotype.\n\n== SUMMARY ==\nTotal reads processed:                     25922\nReading /dev/fd/63\n... read 2133 records:\n...   2133 reads are assigned to a haplotype.\n\n== SUMMARY ==\nTotal reads processed:                     25922\nReading /dev/fd/63\n... read 4839 records:\n...   4839 reads are assigned to a haplotype.\n\n== SUMMARY ==\nTotal reads processed:                     25922\nReading /dev/fd/63\n... read 7290 records:\n...   7290 reads are assigned to a haplotype.\n\n== SUMMARY ==\nTotal reads processed:                     15421\nReading /dev/fd/63\n... read 30 records:\n...   30 reads are assigned to a haplotype.\n\n== SUMMARY ==\nTotal reads processed:                     15421\nReading /dev/fd/63\n... read 1049 records:\n...   1049 reads are assigned to a haplotype.\n\n== SUMMARY ==\nTotal reads processed:                     15421\nReading /dev/fd/63\n... read 2398 records:\n...   2398 reads are assigned to a haplotype.\n\n== SUMMARY ==\nTotal reads processed:                     15421\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "%%bash\n\n\nbash ./split_fastq_w_untagged.sh \\\nreads/HG002.Sequel.15kb.pbmm2.hs37d5.whatshap.haplotag.RTG.10x.trio.MHConly.fastq.gz \\\ncram/HG002.PacBio.15kbCCS.sorted.tags-newsplit.tsv\n", "metadata": {"trusted": true}, "execution_count": 32, "outputs": [{"name": "stdout", "text": "Block 28498559\n", "output_type": "stream"}, {"name": "stderr", "text": "Reading /dev/fd/63\n... read 22389 records:\n...   22389 reads are assigned to a haplotype.\n\n== SUMMARY ==\nTotal reads processed:                     25922\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "%%bash\ndx upload --recursive reads/ --path /phased_reads_2/", "metadata": {"trusted": true}, "execution_count": 33, "outputs": []}, {"cell_type": "code", "source": "", "metadata": {}, "execution_count": null, "outputs": []}]}