{"metadata": {"language_info": {"name": "python", "version": "3.7.3", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "pygments_lexer": "ipython3", "nbconvert_exporter": "python", "file_extension": ".py"}, "kernelspec": {"name": "python3", "display_name": "Python 3", "language": "python"}}, "nbformat_minor": 4, "nbformat": 4, "cells": [{"cell_type": "code", "source": "%%capture\n%%bash\n. /opt/conda/bin/activate\nconda install -c bioconda -y samtools\nconda install -c bioconda -y tabix\nconda install -c bioconda -y pbmm2 pbsv", "metadata": {"trusted": true}, "execution_count": 1, "outputs": []}, {"cell_type": "code", "source": "%%capture\n%%bash\ndx download -f --recursive /Haplotype_ASM_0525_2019/\nln -sf Haplotype_ASM_0525_2019/* .", "metadata": {"trusted": true}, "execution_count": 2, "outputs": []}, {"cell_type": "code", "source": "%%bash\ndx download -f /jc_notebook/MHC_37.fa ", "metadata": {"trusted": true}, "execution_count": 3, "outputs": []}, {"cell_type": "code", "source": "%%capture\n%%bash\ngit clone https://github.com/lh3/minimap2\ncd minimap2 && make", "metadata": {"trusted": true}, "execution_count": 4, "outputs": []}, {"cell_type": "code", "source": "%%capture\n%%bash\ncurl -L https://github.com/attractivechaos/k8/releases/download/v0.2.4/k8-0.2.4.tar.bz2 | tar -jxf -\ncp k8-0.2.4/k8-`uname -s` k8   # or copy it to a directory on you $PATH", "metadata": {"trusted": true}, "execution_count": 5, "outputs": []}, {"cell_type": "code", "source": "%%bash\ncp MHC_37.fa ref.fa", "metadata": {"trusted": true}, "execution_count": 6, "outputs": []}, {"cell_type": "code", "source": "%%bash\nminimap2/minimap2 -c --cs ref.fa p_ctg_cns_H1.fa | \\\nsort -k6,6 -k8,8n  | \\\n./k8 minimap2/misc/paftools.js call -f ref.fa  -L200 - > H1toMHC_var.vcf\nbgzip -f H1toMHC_var.vcf\ntabix -p vcf -f H1toMHC_var.vcf.gz", "metadata": {"trusted": true}, "execution_count": 7, "outputs": [{"name": "stderr", "text": "[M::mm_idx_gen::0.156*1.00] collected minimizers\n[M::mm_idx_gen::0.192*1.33] sorted minimizers\n[M::main::0.192*1.33] loaded/built the index for 1 target sequence(s)\n[M::mm_mapopt_update::0.206*1.30] mid_occ = 99\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 1\n[M::mm_idx_stat::0.216*1.28] distinct minimizers: 768040 (92.97% are singletons); average occurrences: 1.219; average spacing: 5.307\n[M::worker_pipeline::12.086*1.04] mapped 7 sequences\n[M::main] Version: 2.17-r954-dirty\n[M::main] CMD: minimap2/minimap2 -c --cs ref.fa p_ctg_cns_H1.fa\n[M::main] Real time: 12.095 sec; CPU: 12.564 sec; Peak RSS: 0.239 GB\n4861902 reference bases covered by exactly one contig\n16457 substitutions; ts/tv = 2.119\n625 1bp deletions\n537 1bp insertions\n262 2bp deletions\n230 2bp insertions\n451 [3,50) deletions\n399 [3,50) insertions\n24 [50,1000) deletions\n23 [50,1000) insertions\n3 >=1000 deletions\n1 >=1000 insertions\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "%%bash\nminimap2/minimap2 -c --cs ref.fa p_ctg_cns_H2.fa | \\\nsort -k6,6 -k8,8n  | \\\n./k8 minimap2/misc/paftools.js call -f ref.fa  -L200 - > H2toMHC_var.vcf\nbgzip -f H2toMHC_var.vcf\ntabix -p vcf -f H2toMHC_var.vcf.gz", "metadata": {"trusted": true}, "execution_count": 8, "outputs": [{"name": "stderr", "text": "[M::mm_idx_gen::0.155*0.98] collected minimizers\n[M::mm_idx_gen::0.194*1.38] sorted minimizers\n[M::main::0.194*1.38] loaded/built the index for 1 target sequence(s)\n[M::mm_mapopt_update::0.208*1.37] mid_occ = 99\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 1\n[M::mm_idx_stat::0.217*1.34] distinct minimizers: 768040 (92.97% are singletons); average occurrences: 1.219; average spacing: 5.307\n[M::worker_pipeline::12.703*1.02] mapped 10 sequences\n[M::main] Version: 2.17-r954-dirty\n[M::main] CMD: minimap2/minimap2 -c --cs ref.fa p_ctg_cns_H2.fa\n[M::main] Real time: 12.712 sec; CPU: 13.004 sec; Peak RSS: 0.206 GB\n4847120 reference bases covered by exactly one contig\n17832 substitutions; ts/tv = 2.152\n583 1bp deletions\n598 1bp insertions\n226 2bp deletions\n231 2bp insertions\n437 [3,50) deletions\n445 [3,50) insertions\n24 [50,1000) deletions\n32 [50,1000) insertions\n5 >=1000 deletions\n3 >=1000 insertions\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "%%bash\ndx download -f --recursive /phased_reads_2/", "metadata": {"trusted": true}, "execution_count": 9, "outputs": []}, {"cell_type": "code", "source": "%%bash\ngit clone https://github.com/lh3/seqtk.git\ncd seqtk; make", "metadata": {"trusted": true}, "execution_count": 10, "outputs": [{"name": "stdout", "text": "gcc -g -Wall -O2 -Wno-unused-function seqtk.c -o seqtk -lz -lm\n", "output_type": "stream"}, {"name": "stderr", "text": "Cloning into 'seqtk'...\nseqtk.c: In function \u2018stk_comp\u2019:\nseqtk.c:399:16: warning: variable \u2018lc\u2019 set but not used [-Wunused-but-set-variable]\n    int la, lb, lc, na, nb, nc, cnt[11];\n                ^~\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "%%bash\nzcat phased_reads_2/HG002.Sequel.15kb.pbmm2.hs37d5.whatshap.haplotag.RTG.10x.trio.MHConly.fastq.gz | awk 'NR % 4 ==1' | tr -d \"@\" | sort > ids1\ncat phased_reads_2/asm_mapped.fa | grep \">\" | tr -d \">\" | sort > ids2\ncomm  -13 ids1 ids2  > unphased_reads_on_assembly_ids\n./seqtk/seqtk subseq phased_reads_2/asm_mapped.fa unphased_reads_on_assembly_ids > unphased_reads_on_assembly.fa", "metadata": {"trusted": true}, "execution_count": 11, "outputs": []}, {"cell_type": "code", "source": "%%bash\nseqtk/seqtk seq -A phased_reads_2/HG002.Sequel.15kb.pbmm2.hs37d5.whatshap.haplotag.RTG.10x.trio.MHConly.28498559.H1.fastq.gz > reads_H1.fa\nseqtk/seqtk seq -A phased_reads_2/HG002.Sequel.15kb.pbmm2.hs37d5.whatshap.haplotag.RTG.10x.trio.MHConly.28498559.untagged.fastq.gz >> reads_H1.fa\ncat unphased_reads_on_assembly.fa >> reads_H1.fa", "metadata": {"trusted": true}, "execution_count": 12, "outputs": []}, {"cell_type": "code", "source": "%%bash\nseqtk/seqtk seq -A phased_reads_2/HG002.Sequel.15kb.pbmm2.hs37d5.whatshap.haplotag.RTG.10x.trio.MHConly.28498559.H2.fastq.gz > reads_H2.fa\nseqtk/seqtk seq -A phased_reads_2/HG002.Sequel.15kb.pbmm2.hs37d5.whatshap.haplotag.RTG.10x.trio.MHConly.28498559.untagged.fastq.gz >> reads_H2.fa\ncat unphased_reads_on_assembly.fa >> reads_H2.fa", "metadata": {"trusted": true}, "execution_count": 13, "outputs": []}, {"cell_type": "code", "source": "%%bash\nminimap2/minimap2 -a -x asm5 ref.fa reads_H1.fa | samtools view -bS - | samtools sort - > reads_H1_MHC.bam \nsamtools index reads_H1_MHC.bam\nminimap2/minimap2 -a -x asm5 ref.fa reads_H2.fa | samtools view -bS - | samtools sort - > reads_H2_MHC.bam \nsamtools index reads_H2_MHC.bam", "metadata": {"trusted": true}, "execution_count": 14, "outputs": [{"name": "stderr", "text": "[M::mm_idx_gen::0.124*1.00] collected minimizers\n[M::mm_idx_gen::0.143*1.23] sorted minimizers\n[M::main::0.143*1.23] loaded/built the index for 1 target sequence(s)\n[M::mm_mapopt_update::0.151*1.22] mid_occ = 100\n[M::mm_idx_stat] kmer size: 19; skip: 19; is_hpc: 0; #seq: 1\n[M::mm_idx_stat::0.156*1.23] distinct minimizers: 438505 (95.36% are singletons); average occurrences: 1.139; average spacing: 9.953\n[M::worker_pipeline::22.801*2.67] mapped 14990 sequences\n[M::main] Version: 2.17-r954-dirty\n[M::main] CMD: minimap2/minimap2 -a -x asm5 ref.fa reads_H1.fa\n[M::main] Real time: 22.815 sec; CPU: 60.932 sec; Peak RSS: 0.421 GB\n[M::mm_idx_gen::0.125*0.96] collected minimizers\n[M::mm_idx_gen::0.145*1.24] sorted minimizers\n[M::main::0.145*1.24] loaded/built the index for 1 target sequence(s)\n[M::mm_mapopt_update::0.153*1.23] mid_occ = 100\n[M::mm_idx_stat] kmer size: 19; skip: 19; is_hpc: 0; #seq: 1\n[M::mm_idx_stat::0.158*1.21] distinct minimizers: 438505 (95.36% are singletons); average occurrences: 1.139; average spacing: 9.953\n[M::worker_pipeline::23.782*2.70] mapped 14823 sequences\n[M::main] Version: 2.17-r954-dirty\n[M::main] CMD: minimap2/minimap2 -a -x asm5 ref.fa reads_H2.fa\n[M::main] Real time: 23.795 sec; CPU: 64.120 sec; Peak RSS: 0.507 GB\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "%%bash\npbmm2 align ref.fa reads_H1.fa --sort --preset CCS --sample sample1 --rg '@RG\\tID:movie1' reads_H1_MHC.bam\npbmm2 align ref.fa reads_H2.fa --sort --preset CCS --sample sample1 --rg '@RG\\tID:movie1' reads_H2_MHC.bam", "metadata": {"trusted": true}, "execution_count": 15, "outputs": [{"name": "stderr", "text": ">|> 20190903 19:56:59.007 -|- WARN -|- CheckPositionalArgs -|- 0x7fa6ce8e3b80|| -|- Input is FASTA. Output BAM file cannot be used for polishing with GenomicConsensus!\n>|> 20190903 19:56:59.007 -|- WARN -|- CheckPositionalArgs -|- 0x7fa6ce8e3b80|| -|- Warning: Overwriting existing output file: reads_H1_MHC.bam\n>|> 20190903 19:57:17.447 -|- WARN -|- CheckPositionalArgs -|- 0x7eff4e1deb80|| -|- Input is FASTA. Output BAM file cannot be used for polishing with GenomicConsensus!\n>|> 20190903 19:57:17.447 -|- WARN -|- CheckPositionalArgs -|- 0x7eff4e1deb80|| -|- Warning: Overwriting existing output file: reads_H2_MHC.bam\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "%%bash\nsamtools faidx ref.fa\ndx mkdir -p /tmp_data_2\ndx cd /tmp_data_2\ndx upload ref.fa*", "metadata": {"trusted": true}, "execution_count": 16, "outputs": [{"name": "stdout", "text": "ID                  file-FbgGFG80Xv397KfJKb6j3xf2\nClass               file\nProject             project-FXGVFb00Xv39f6K624b11bZ1\nFolder              /tmp_data_2\nName                ref.fa\nState               closing\nVisibility          visible\nTypes               -\nProperties          -\nTags                -\nOutgoing links      -\nCreated             Tue Sep  3 12:57:37 2019\nCreated by          jchin_dx\n via the job        job-Fbg8Jk80Xv3F7zbyKvk80ZJp\nLast modified       Tue Sep  3 12:57:37 2019\nMedia type          \narchivalState       \"live\"\nID                  file-FbgGFG80Xv3531ZP2jjf4Jq3\nClass               file\nProject             project-FXGVFb00Xv39f6K624b11bZ1\nFolder              /tmp_data_2\nName                ref.fa.fai\nState               closing\nVisibility          visible\nTypes               -\nProperties          -\nTags                -\nOutgoing links      -\nCreated             Tue Sep  3 12:57:37 2019\nCreated by          jchin_dx\n via the job        job-Fbg8Jk80Xv3F7zbyKvk80ZJp\nLast modified       Tue Sep  3 12:57:38 2019\nMedia type          \narchivalState       \"live\"\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "%%bash\ndx mkdir -p /tmp_data_2\ndx cd /tmp_data_2\ndx upload reads_H2_MHC.bam* reads_H1_MHC.bam* H2toMHC_var.vcf* H1toMHC_var.vcf*", "metadata": {"trusted": true}, "execution_count": 17, "outputs": [{"name": "stdout", "text": "ID                  file-FbgGFGj0Xv35YYZv2jY144bP\nClass               file\nProject             project-FXGVFb00Xv39f6K624b11bZ1\nFolder              /tmp_data_2\nName                reads_H2_MHC.bam\nState               closing\nVisibility          visible\nTypes               -\nProperties          -\nTags                -\nOutgoing links      -\nCreated             Tue Sep  3 12:57:39 2019\nCreated by          jchin_dx\n via the job        job-Fbg8Jk80Xv3F7zbyKvk80ZJp\nLast modified       Tue Sep  3 12:57:40 2019\nMedia type          \narchivalState       \"live\"\nID                  file-FbgGFJ00Xv3KVp084QQZyyyy\nClass               file\nProject             project-FXGVFb00Xv39f6K624b11bZ1\nFolder              /tmp_data_2\nName                reads_H2_MHC.bam.bai\nState               closing\nVisibility          visible\nTypes               -\nProperties          -\nTags                -\nOutgoing links      -\nCreated             Tue Sep  3 12:57:40 2019\nCreated by          jchin_dx\n via the job        job-Fbg8Jk80Xv3F7zbyKvk80ZJp\nLast modified       Tue Sep  3 12:57:40 2019\nMedia type          \narchivalState       \"live\"\nID                  file-FbgGFJ00Xv3KVp084QQZyyz0\nClass               file\nProject             project-FXGVFb00Xv39f6K624b11bZ1\nFolder              /tmp_data_2\nName                reads_H1_MHC.bam\nState               closing\nVisibility          visible\nTypes               -\nProperties          -\nTags                -\nOutgoing links      -\nCreated             Tue Sep  3 12:57:40 2019\nCreated by          jchin_dx\n via the job        job-Fbg8Jk80Xv3F7zbyKvk80ZJp\nLast modified       Tue Sep  3 12:57:42 2019\nMedia type          \narchivalState       \"live\"\nID                  file-FbgGFJQ0Xv3KQ4f6KbgZX6FY\nClass               file\nProject             project-FXGVFb00Xv39f6K624b11bZ1\nFolder              /tmp_data_2\nName                reads_H1_MHC.bam.bai\nState               closing\nVisibility          visible\nTypes               -\nProperties          -\nTags                -\nOutgoing links      -\nCreated             Tue Sep  3 12:57:42 2019\nCreated by          jchin_dx\n via the job        job-Fbg8Jk80Xv3F7zbyKvk80ZJp\nLast modified       Tue Sep  3 12:57:42 2019\nMedia type          \narchivalState       \"live\"\nID                  file-FbgGFJj0Xv3GYbJJ2k2Bp3V3\nClass               file\nProject             project-FXGVFb00Xv39f6K624b11bZ1\nFolder              /tmp_data_2\nName                H2toMHC_var.vcf.gz\nState               closing\nVisibility          visible\nTypes               -\nProperties          -\nTags                -\nOutgoing links      -\nCreated             Tue Sep  3 12:57:43 2019\nCreated by          jchin_dx\n via the job        job-Fbg8Jk80Xv3F7zbyKvk80ZJp\nLast modified       Tue Sep  3 12:57:43 2019\nMedia type          \narchivalState       \"live\"\nID                  file-FbgGFJj0Xv3J7F2F2k9yfq20\nClass               file\nProject             project-FXGVFb00Xv39f6K624b11bZ1\nFolder              /tmp_data_2\nName                H2toMHC_var.vcf.gz.tbi\nState               closing\nVisibility          visible\nTypes               -\nProperties          -\nTags                -\nOutgoing links      -\nCreated             Tue Sep  3 12:57:43 2019\nCreated by          jchin_dx\n via the job        job-Fbg8Jk80Xv3F7zbyKvk80ZJp\nLast modified       Tue Sep  3 12:57:44 2019\nMedia type          \narchivalState       \"live\"\nID                  file-FbgGFK00Xv3JbP7V2jK9V149\nClass               file\nProject             project-FXGVFb00Xv39f6K624b11bZ1\nFolder              /tmp_data_2\nName                H1toMHC_var.vcf.gz\nState               closing\nVisibility          visible\nTypes               -\nProperties          -\nTags                -\nOutgoing links      -\nCreated             Tue Sep  3 12:57:44 2019\nCreated by          jchin_dx\n via the job        job-Fbg8Jk80Xv3F7zbyKvk80ZJp\nLast modified       Tue Sep  3 12:57:44 2019\nMedia type          \narchivalState       \"live\"\nID                  file-FbgGFK80Xv3973ZkKf6q7B3k\nClass               file\nProject             project-FXGVFb00Xv39f6K624b11bZ1\nFolder              /tmp_data_2\nName                H1toMHC_var.vcf.gz.tbi\nState               closing\nVisibility          visible\nTypes               -\nProperties          -\nTags                -\nOutgoing links      -\nCreated             Tue Sep  3 12:57:45 2019\nCreated by          jchin_dx\n via the job        job-Fbg8Jk80Xv3F7zbyKvk80ZJp\nLast modified       Tue Sep  3 12:57:45 2019\nMedia type          \narchivalState       \"live\"\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "%%bash\npbsv discover reads_H1_MHC.bam reads_H1_MHC.svsig.gz \npbsv discover reads_H2_MHC.bam reads_H2_MHC.svsig.gz ", "metadata": {"trusted": true}, "execution_count": 18, "outputs": []}, {"cell_type": "code", "source": "%%bash\npbsv call --ccs ref.fa reads_H1_MHC.svsig.gz reads_H1_MHC.var.vcf\npbsv call --ccs ref.fa reads_H2_MHC.svsig.gz reads_H2_MHC.var.vcf\nbgzip -f reads_H1_MHC.var.vcf \nbgzip -f reads_H2_MHC.var.vcf\ntabix -p vcf -f reads_H1_MHC.var.vcf.gz\ntabix -p vcf -f reads_H2_MHC.var.vcf.gz", "metadata": {"trusted": true}, "execution_count": 19, "outputs": []}, {"cell_type": "code", "source": "%%bash\ndx mkdir -p /tmp_data_2\ndx cd /tmp_data_2\ndx upload reads_H1_MHC.var.vcf* reads_H2_MHC.var.vcf*", "metadata": {"trusted": true}, "execution_count": 20, "outputs": [{"name": "stdout", "text": "ID                  file-FbgGFf80Xv3077q6Kb9VByZJ\nClass               file\nProject             project-FXGVFb00Xv39f6K624b11bZ1\nFolder              /tmp_data_2\nName                reads_H1_MHC.var.vcf.gz\nState               closing\nVisibility          visible\nTypes               -\nProperties          -\nTags                -\nOutgoing links      -\nCreated             Tue Sep  3 12:58:17 2019\nCreated by          jchin_dx\n via the job        job-Fbg8Jk80Xv3F7zbyKvk80ZJp\nLast modified       Tue Sep  3 12:58:17 2019\nMedia type          \narchivalState       \"live\"\nID                  file-FbgGFf80Xv3Jy2byKbPJ4yVp\nClass               file\nProject             project-FXGVFb00Xv39f6K624b11bZ1\nFolder              /tmp_data_2\nName                reads_H1_MHC.var.vcf.gz.tbi\nState               closing\nVisibility          visible\nTypes               -\nProperties          -\nTags                -\nOutgoing links      -\nCreated             Tue Sep  3 12:58:17 2019\nCreated by          jchin_dx\n via the job        job-Fbg8Jk80Xv3F7zbyKvk80ZJp\nLast modified       Tue Sep  3 12:58:18 2019\nMedia type          \narchivalState       \"live\"\nID                  file-FbgGFfQ0Xv35YYZv2jY144bk\nClass               file\nProject             project-FXGVFb00Xv39f6K624b11bZ1\nFolder              /tmp_data_2\nName                reads_H2_MHC.var.vcf.gz\nState               closing\nVisibility          visible\nTypes               -\nProperties          -\nTags                -\nOutgoing links      -\nCreated             Tue Sep  3 12:58:18 2019\nCreated by          jchin_dx\n via the job        job-Fbg8Jk80Xv3F7zbyKvk80ZJp\nLast modified       Tue Sep  3 12:58:18 2019\nMedia type          \narchivalState       \"live\"\nID                  file-FbgGFfQ0Xv31KvJ84PKVGzQK\nClass               file\nProject             project-FXGVFb00Xv39f6K624b11bZ1\nFolder              /tmp_data_2\nName                reads_H2_MHC.var.vcf.gz.tbi\nState               closing\nVisibility          visible\nTypes               -\nProperties          -\nTags                -\nOutgoing links      -\nCreated             Tue Sep  3 12:58:18 2019\nCreated by          jchin_dx\n via the job        job-Fbg8Jk80Xv3F7zbyKvk80ZJp\nLast modified       Tue Sep  3 12:58:19 2019\nMedia type          \narchivalState       \"live\"\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "%%bash\n\npbmm2 align p_ctg_cns_H1.fa reads_H1.fa --sort --preset CCS --sample sample1 --rg '@RG\\tID:H1' reads_H1.bam\npbmm2 align p_ctg_cns_H2.fa reads_H2.fa --sort --preset CCS --sample sample1 --rg '@RG\\tID:H2' reads_H2.bam\npbsv discover reads_H1.bam reads_H1.svsig.gz \npbsv discover reads_H2.bam reads_H2.svsig.gz \npbsv call --ccs p_ctg_cns_H1.fa reads_H1.svsig.gz reads_H1.var.vcf\npbsv call --ccs p_ctg_cns_H2.fa reads_H2.svsig.gz reads_H2.var.vcf\nbgzip -f reads_H1.var.vcf \nbgzip -f reads_H2.var.vcf\ntabix -p vcf -f reads_H1.var.vcf.gz\ntabix -p vcf -f reads_H2.var.vcf.gz", "metadata": {"trusted": true}, "execution_count": 21, "outputs": [{"name": "stderr", "text": ">|> 20190903 19:58:19.580 -|- WARN -|- CheckPositionalArgs -|- 0x7f3a9c3d6b80|| -|- Input is FASTA. Output BAM file cannot be used for polishing with GenomicConsensus!\n>|> 20190903 19:58:34.963 -|- WARN -|- CheckPositionalArgs -|- 0x7f988a411b80|| -|- Input is FASTA. Output BAM file cannot be used for polishing with GenomicConsensus!\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "%%bash\ndx mkdir -p /tmp_data_2\ndx cd /tmp_data_2\ndx upload reads_H?.var.vcf.gz* reads_H?.bam* p_ctg_cns_H?.fa", "metadata": {"trusted": true}, "execution_count": 22, "outputs": [{"name": "stdout", "text": "ID                  file-FbgGFzQ0Xv3Kbk44Kf327Xk3\nClass               file\nProject             project-FXGVFb00Xv39f6K624b11bZ1\nFolder              /tmp_data_2\nName                reads_H1.var.vcf.gz\nState               closing\nVisibility          visible\nTypes               -\nProperties          -\nTags                -\nOutgoing links      -\nCreated             Tue Sep  3 12:58:54 2019\nCreated by          jchin_dx\n via the job        job-Fbg8Jk80Xv3F7zbyKvk80ZJp\nLast modified       Tue Sep  3 12:58:54 2019\nMedia type          \narchivalState       \"live\"\nID                  file-FbgGFzQ0Xv3Kbk44Kf327Xk5\nClass               file\nProject             project-FXGVFb00Xv39f6K624b11bZ1\nFolder              /tmp_data_2\nName                reads_H1.var.vcf.gz.tbi\nState               closing\nVisibility          visible\nTypes               -\nProperties          -\nTags                -\nOutgoing links      -\nCreated             Tue Sep  3 12:58:54 2019\nCreated by          jchin_dx\n via the job        job-Fbg8Jk80Xv3F7zbyKvk80ZJp\nLast modified       Tue Sep  3 12:58:55 2019\nMedia type          \narchivalState       \"live\"\nID                  file-FbgGFzj0Xv3077q6Kb9VByb1\nClass               file\nProject             project-FXGVFb00Xv39f6K624b11bZ1\nFolder              /tmp_data_2\nName                reads_H2.var.vcf.gz\nState               closing\nVisibility          visible\nTypes               -\nProperties          -\nTags                -\nOutgoing links      -\nCreated             Tue Sep  3 12:58:55 2019\nCreated by          jchin_dx\n via the job        job-Fbg8Jk80Xv3F7zbyKvk80ZJp\nLast modified       Tue Sep  3 12:58:55 2019\nMedia type          \narchivalState       \"live\"\nID                  file-FbgGG000Xv328Z9z2jykG6Jk\nClass               file\nProject             project-FXGVFb00Xv39f6K624b11bZ1\nFolder              /tmp_data_2\nName                reads_H2.var.vcf.gz.tbi\nState               closing\nVisibility          visible\nTypes               -\nProperties          -\nTags                -\nOutgoing links      -\nCreated             Tue Sep  3 12:58:56 2019\nCreated by          jchin_dx\n via the job        job-Fbg8Jk80Xv3F7zbyKvk80ZJp\nLast modified       Tue Sep  3 12:58:56 2019\nMedia type          \narchivalState       \"live\"\nID                  file-FbgGG000Xv3Kbk44Kf327Xk7\nClass               file\nProject             project-FXGVFb00Xv39f6K624b11bZ1\nFolder              /tmp_data_2\nName                reads_H1.bam\nState               closing\nVisibility          visible\nTypes               -\nProperties          -\nTags                -\nOutgoing links      -\nCreated             Tue Sep  3 12:58:56 2019\nCreated by          jchin_dx\n via the job        job-Fbg8Jk80Xv3F7zbyKvk80ZJp\nLast modified       Tue Sep  3 12:58:57 2019\nMedia type          \narchivalState       \"live\"\nID                  file-FbgGG080Xv34B184KbXBpYq5\nClass               file\nProject             project-FXGVFb00Xv39f6K624b11bZ1\nFolder              /tmp_data_2\nName                reads_H1.bam.bai\nState               closing\nVisibility          visible\nTypes               -\nProperties          -\nTags                -\nOutgoing links      -\nCreated             Tue Sep  3 12:58:57 2019\nCreated by          jchin_dx\n via the job        job-Fbg8Jk80Xv3F7zbyKvk80ZJp\nLast modified       Tue Sep  3 12:58:58 2019\nMedia type          \narchivalState       \"live\"\nID                  file-FbgGG0Q0Xv31KvJ84PKVGzQv\nClass               file\nProject             project-FXGVFb00Xv39f6K624b11bZ1\nFolder              /tmp_data_2\nName                reads_H2.bam\nState               closing\nVisibility          visible\nTypes               -\nProperties          -\nTags                -\nOutgoing links      -\nCreated             Tue Sep  3 12:58:58 2019\nCreated by          jchin_dx\n via the job        job-Fbg8Jk80Xv3F7zbyKvk80ZJp\nLast modified       Tue Sep  3 12:58:59 2019\nMedia type          \narchivalState       \"live\"\nID                  file-FbgGG0j0Xv3Jy2byKbPJ4yX6\nClass               file\nProject             project-FXGVFb00Xv39f6K624b11bZ1\nFolder              /tmp_data_2\nName                reads_H2.bam.bai\nState               closing\nVisibility          visible\nTypes               -\nProperties          -\nTags                -\nOutgoing links      -\nCreated             Tue Sep  3 12:58:59 2019\nCreated by          jchin_dx\n via the job        job-Fbg8Jk80Xv3F7zbyKvk80ZJp\nLast modified       Tue Sep  3 12:59:00 2019\nMedia type          \narchivalState       \"live\"\nID                  file-FbgGG100Xv397KfJKb6j3xgJ\nClass               file\nProject             project-FXGVFb00Xv39f6K624b11bZ1\nFolder              /tmp_data_2\nName                p_ctg_cns_H1.fa\nState               closing\nVisibility          visible\nTypes               -\nProperties          -\nTags                -\nOutgoing links      -\nCreated             Tue Sep  3 12:59:00 2019\nCreated by          jchin_dx\n via the job        job-Fbg8Jk80Xv3F7zbyKvk80ZJp\nLast modified       Tue Sep  3 12:59:01 2019\nMedia type          \narchivalState       \"live\"\nID                  file-FbgGG180Xv3JbP7V2jK9V157\nClass               file\nProject             project-FXGVFb00Xv39f6K624b11bZ1\nFolder              /tmp_data_2\nName                p_ctg_cns_H2.fa\nState               closing\nVisibility          visible\nTypes               -\nProperties          -\nTags                -\nOutgoing links      -\nCreated             Tue Sep  3 12:59:01 2019\nCreated by          jchin_dx\n via the job        job-Fbg8Jk80Xv3F7zbyKvk80ZJp\nLast modified       Tue Sep  3 12:59:02 2019\nMedia type          \narchivalState       \"live\"\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "%%writefile readme.txt\n#assembly to assembly SV call by pfatools\nH1toMHC_var.vcf.gz\nH1toMHC_var.vcf.gz.tbi\nH2toMHC_var.vcf.gz\nH2toMHC_var.vcf.gz.tbi\n\n#notebook on how the variants are called\nMHC_Analysis_00.ipynb\n\n# assembly used\np_ctg_cns_H1.fa\np_ctg_cns_H2.fa\n\n#this file\nreadme.txt\n\n#reads to assembly variant call\nreads_H1.bam\nreads_H1.bam.bai\nreads_H1.var.vcf.gz\nreads_H1.var.vcf.gz.tbi\nreads_H2.bam\nreads_H2.bam.bai\nreads_H2.var.vcf.gz\nreads_H2.var.vcf.gz.tbi\n\n#reads to GRCh37 MHC variant call\nreads_H1_MHC.bam\nreads_H1_MHC.bam.bai\nreads_H1_MHC.var.vcf.gz\nreads_H1_MHC.var.vcf.gz.tbi\nreads_H2_MHC.bam\nreads_H2_MHC.bam.bai\nreads_H2_MHC.var.vcf.gz\nreads_H2_MHC.var.vcf.gz.tbi\n\n#GRCh 37 MHC seqeunce\nref.fa\nref.fa.fai", "metadata": {"trusted": true}, "execution_count": 23, "outputs": [{"name": "stdout", "text": "Writing readme.txt\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "%%bash\ndx mkdir -p /tmp_data_2\ndx cd /tmp_data_2\ndx upload readme.txt", "metadata": {"trusted": true}, "execution_count": 24, "outputs": [{"name": "stdout", "text": "ID                  file-FbgGG1j0Xv3GYbJJ2k2Bp3X7\nClass               file\nProject             project-FXGVFb00Xv39f6K624b11bZ1\nFolder              /tmp_data_2\nName                readme.txt\nState               closing\nVisibility          visible\nTypes               -\nProperties          -\nTags                -\nOutgoing links      -\nCreated             Tue Sep  3 12:59:03 2019\nCreated by          jchin_dx\n via the job        job-Fbg8Jk80Xv3F7zbyKvk80ZJp\nLast modified       Tue Sep  3 12:59:03 2019\nMedia type          \narchivalState       \"live\"\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "%%bash\nseqtk/seqtk seq -A phased_reads_2/HG002.Sequel.15kb.pbmm2.hs37d5.whatshap.haplotag.RTG.10x.trio.MHConly.28498559.H1.fastq.gz > all_reads.fa\nseqtk/seqtk seq -A phased_reads_2/HG002.Sequel.15kb.pbmm2.hs37d5.whatshap.haplotag.RTG.10x.trio.MHConly.28498559.H2.fastq.gz > all_reads.fa\nseqtk/seqtk seq -A phased_reads_2/HG002.Sequel.15kb.pbmm2.hs37d5.whatshap.haplotag.RTG.10x.trio.MHConly.28498559.untagged.fastq.gz >> all_reads.fa\ncat unphased_reads_on_assembly.fa >> rall_reads.fa\n\n\npbmm2 align ref.fa all_reads.fa --sort --preset CCS --sample sample1 --rg '@RG\\tID:movie1' all_CCS_reads_MHC.bam\n\npbsv discover all_CCS_reads_MHC.bam all_CCS_reads_MHC.svsig.gz \npbsv call --ccs MHC_37.fa all_CCS_reads_MHC.svsig.gz all_CCS_reads_MHC.var.vcf\nbgzip -f all_CCS_reads_MHC.var.vcf\ntabix -p vcf -f all_CCS_reads_MHC.var.vcf.gz\n", "metadata": {"trusted": true}, "execution_count": 25, "outputs": [{"name": "stderr", "text": ">|> 20190903 19:59:06.997 -|- WARN -|- CheckPositionalArgs -|- 0x7f5437893b80|| -|- Input is FASTA. Output BAM file cannot be used for polishing with GenomicConsensus!\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "%%bash\ndx mkdir -p /tmp_data_2\ndx cd /tmp_data_2\ndx upload all_CCS_reads_MHC.var.vcf.gz all_CCS_reads_MHC.var.vcf.gz.tbi  all_CCS_reads_MHC.bam  all_CCS_reads_MHC.bam.bai", "metadata": {"trusted": true}, "execution_count": 26, "outputs": [{"name": "stdout", "text": "ID                  file-FbgGGB80Xv3Kjz2P4Pj52b3J\nClass               file\nProject             project-FXGVFb00Xv39f6K624b11bZ1\nFolder              /tmp_data_2\nName                all_CCS_reads_MHC.var.vcf.gz\nState               closing\nVisibility          visible\nTypes               -\nProperties          -\nTags                -\nOutgoing links      -\nCreated             Tue Sep  3 12:59:37 2019\nCreated by          jchin_dx\n via the job        job-Fbg8Jk80Xv3F7zbyKvk80ZJp\nLast modified       Tue Sep  3 12:59:37 2019\nMedia type          \narchivalState       \"live\"\nID                  file-FbgGGB80Xv3KVp084QQZyz0K\nClass               file\nProject             project-FXGVFb00Xv39f6K624b11bZ1\nFolder              /tmp_data_2\nName                all_CCS_reads_MHC.var.vcf.gz.tbi\nState               closing\nVisibility          visible\nTypes               -\nProperties          -\nTags                -\nOutgoing links      -\nCreated             Tue Sep  3 12:59:37 2019\nCreated by          jchin_dx\n via the job        job-Fbg8Jk80Xv3F7zbyKvk80ZJp\nLast modified       Tue Sep  3 12:59:38 2019\nMedia type          \narchivalState       \"live\"\nID                  file-FbgGGBQ0Xv31KvJ84PKVGzV6\nClass               file\nProject             project-FXGVFb00Xv39f6K624b11bZ1\nFolder              /tmp_data_2\nName                all_CCS_reads_MHC.bam\nState               closing\nVisibility          visible\nTypes               -\nProperties          -\nTags                -\nOutgoing links      -\nCreated             Tue Sep  3 12:59:38 2019\nCreated by          jchin_dx\n via the job        job-Fbg8Jk80Xv3F7zbyKvk80ZJp\nLast modified       Tue Sep  3 12:59:39 2019\nMedia type          \narchivalState       \"live\"\nID                  file-FbgGGBj0Xv3KVp084QQZyz0Q\nClass               file\nProject             project-FXGVFb00Xv39f6K624b11bZ1\nFolder              /tmp_data_2\nName                all_CCS_reads_MHC.bam.bai\nState               closing\nVisibility          visible\nTypes               -\nProperties          -\nTags                -\nOutgoing links      -\nCreated             Tue Sep  3 12:59:39 2019\nCreated by          jchin_dx\n via the job        job-Fbg8Jk80Xv3F7zbyKvk80ZJp\nLast modified       Tue Sep  3 12:59:40 2019\nMedia type          \narchivalState       \"live\"\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "", "metadata": {}, "execution_count": null, "outputs": []}, {"cell_type": "code", "source": "", "metadata": {}, "execution_count": null, "outputs": []}]}