{"metadata": {"language_info": {"name": "python", "version": "3.7.3", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "pygments_lexer": "ipython3", "nbconvert_exporter": "python", "file_extension": ".py"}, "kernelspec": {"name": "python3", "display_name": "Python 3", "language": "python"}}, "nbformat_minor": 4, "nbformat": 4, "cells": [{"cell_type": "code", "source": "%%capture\n%%bash\n. /opt/conda/bin/activat\nconda install -c bioconda freebayes=1.3.1", "metadata": {"trusted": true}, "execution_count": 3, "outputs": []}, {"cell_type": "code", "source": "%%bash\n#freebayes -G 10 -F 0.3 -f p_ctg_cns_H1.fa reads_H1.bam > var_H1.vcf &\n#freebayes -G 10 -F 0.3 -f p_ctg_cns_H2.fa reads_H2.bam > var_H2.vcf &", "metadata": {"trusted": true}, "execution_count": 7, "outputs": [{"name": "stdout", "text": "Process is interrupted.\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "%%bash \n# this takes about one hours to run\nfreebayes-parallel <(fasta_generate_regions.py p_ctg_cns_H1.fa.fai 500000) 8 \\\n   -G 10 -F 0.3  -f p_ctg_cns_H1.fa reads_H1.bam > var_H1_2.vcf &\n\nfreebayes-parallel <(fasta_generate_regions.py p_ctg_cns_H2.fa.fai 500000) 8 \\\n   -G 10 -F 0.3  -f p_ctg_cns_H2.fa reads_H2.bam > var_H2_2.vcf &\n", "metadata": {}, "execution_count": null, "outputs": []}, {"cell_type": "code", "source": "%%bash\nbgzip var_H1_2.vcf\nbgzip var_H2_2.vcf\n", "metadata": {"trusted": true}, "execution_count": 3, "outputs": [{"name": "stderr", "text": "[bgzip] No such file or directory: var_H1_2.vcf\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "%%bash\ntabix -p vcf var_H1_2.vcf.gz\ntabix -p vcf var_H2_2.vcf.gz", "metadata": {"trusted": true}, "execution_count": 6, "outputs": []}, {"cell_type": "code", "source": "import gzip\n\nf = gzip.open(\"var_H1_2.vcf.gz\")\nv_pos = []\nfor row in f:\n    if row[0] == ord(b'#'):\n        continue\n    row = row.strip().split()\n    if row[0] != b'000000F':\n        continue\n    if len(row[3]) != len(row[4]):\n        continue\n    position = int(row[1])\n    v_pos.append(position)\n\nf = gzip.open(\"reads_H1.var.vcf.gz\")\nfor row in f:\n    if row[0] == ord(b'#'):\n        continue\n    row = row.strip().split()\n    if row[0] != b'000000F':\n        continue\n    position = int(row[1])\n    k, v = row[-2:]\n    k = k.split(b':')\n    v = v.split(b':')\n    d = dict(zip(k,v))\n    AD = [int(_) for _ in d[b'AD'].split(b',')]\n    if min(AD)/sum(AD) < 0.25:\n        continue\n    pos = int(row[1])\n    v_pos.append(position)\n\nv_pos.sort()\nSNP_cluster=[]\ncluster_d = 10000\nfor position in v_pos:\n    if len(SNP_cluster) == 0 or position - SNP_cluster[-1][-1] > cluster_d:\n        SNP_cluster.append([position])\n    else:\n        SNP_cluster[-1].append(position)\nfor c in SNP_cluster:\n    #print(\"H1\", len(c), '000000F:{}-{}'.format(min(c)-50, max(c)+50), max(c)-min(c)+100)\n    print('000000F\\t{}\\t{}'.format(min(c)-50, max(c)+50))\nfor c in SNP_cluster:\n    print((min(c)-50, max(c)+50))", "metadata": {"trusted": true}, "execution_count": 14, "outputs": [{"name": "stdout", "text": "000000F\t2241502\t2241602\n000000F\t2389551\t2389651\n000000F\t3475198\t3475399\n000000F\t3486442\t3508043\n000000F\t3956145\t3969225\n(2241502, 2241602)\n(2389551, 2389651)\n(3475198, 3475399)\n(3486442, 3508043)\n(3956145, 3969225)\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "f = gzip.open(\"var_H2_2.vcf.gz\")\nv_pos = []\nfor row in f:\n    if row[0] == ord(b'#'):\n        continue\n    row = row.strip().split()\n    if row[0] != b'000000F':\n        continue\n    if len(row[3]) != len(row[4]):\n        continue\n    position = int(row[1])\n    v_pos.append(position)\n\nf = gzip.open(\"reads_H2.var.vcf.gz\")\nfor row in f:\n    if row[0] == ord(b'#'):\n        continue\n    row = row.strip().split()\n    if row[0] != b'000000F':\n        continue\n    position = int(row[1])\n    k, v = row[-2:]\n    k = k.split(b':')\n    v = v.split(b':')\n    d = dict(zip(k,v))\n    AD = [int(_) for _ in d[b'AD'].split(b',')]\n    if min(AD)/sum(AD) < 0.25:\n        continue\n    pos = int(row[1])\n    v_pos.append(position)\n\nv_pos.sort()\nSNP_cluster=[]\ncluster_d = 10000\nfor position in v_pos:\n    if len(SNP_cluster) == 0 or position - SNP_cluster[-1][-1] > cluster_d:\n        SNP_cluster.append([position])\n    else:\n        SNP_cluster[-1].append(position)\nfor c in SNP_cluster:\n    #print(\"H2\", len(c), '000000F:{}-{}'.format(min(c)-50, max(c)+50), max(c)-min(c)+100)\n    print('000000F\\t{}\\t{}'.format(min(c)-50, max(c)+50))\nfor c in SNP_cluster:\n    print((min(c)-50, max(c)+50))", "metadata": {"trusted": true}, "execution_count": 15, "outputs": [{"name": "stdout", "text": "000000F\t2085306\t2085406\n000000F\t2244234\t2244334\n000000F\t2392104\t2392204\n000000F\t3473257\t3473557\n000000F\t3486094\t3506188\n000000F\t3951022\t3970406\n(2085306, 2085406)\n(2244234, 2244334)\n(2392104, 2392204)\n(3473257, 3473557)\n(3486094, 3506188)\n(3951022, 3970406)\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "", "metadata": {}, "execution_count": null, "outputs": []}]}